### ISSUE TITLE
deployments to uiowa07, uiowa08, uiowa09 test/stage

### ISSUE THREAD

**Issue by briand44** [uiowa/uiowa#8119]
>It looks like test environments > 06 are not being updated when a branch is merged into main. At least I'm not seeing it reflected in [#drupal-status](https://iowaweb.slack.com/archives/CSQA18BK4)

https://iowaweb.slack.com/archives/C0164FNN1PV/p1727709896178109

Hopefully just drush alias issue otherwise something funky with ACN stage vs prod that isn’t triggering it

**Comment by briand44** [comment]
Grooming conversation.. Seems like we just need to dig in to see if we can identify the issue here. We are hopeful it isn't a very big issue and that we can fix as part of this slice. Sized as medium in case we need to create a ticket with Acquia or run into something else unknown.

**Comment by joewhitsitt** [comment]
Capturing from this morning's Slack message:

>here is our code that doesn't appear to be triggering which includes the Slack notification. \Uiowa\Blt\Plugin\Commands\ReplaceCommands::replaceDrupalUpdateAll which I believe overrides this: \Acquia\Blt\Robo\Commands\Artifact\DeployCommand::updateAll and is triggered by \Acquia\Blt\Robo\Commands\Artifact\AcHooksCommand::updateCloudSites. That is as far as I dug previously.

Not sure how to test this (couldn't find a PR change that included test instructions), but also found hooks/test/post-code-update/post-code-update.sh

My guess is there is something wacky with the $target_env as this seems to be the main difference between a cloud classic app and cloud next. (e.g. compare drush/sites/uiowa07.site.yml to drush/sites/uiowa04.site.yml)

**Comment by joewhitsitt** [comment]
i wonder if the scripts in `hooks` could be updated to output variables or other info to a log file that we can then compare after the next merge to main.

or maybe this is an Acquia ticket we should start sooner than later given:

>BLT: Acquia has observed that some applications using BLT may encounter Drush errors or other issues on Acquia Cloud Next environments. Please test all BLT applications and processes immediately after upgrade or provisioning on Acquia Cloud Next to verify that they work as expected. - https://acquia.my.site.com/s/article/1500003132021-Common-Technical-Questions-when-Upgrading-Drupal-Applications-Environments-to-Acquia-Cloud-Next

**Comment by joewhitsitt** [comment]
@e-marie-w here is the stage tasks from the recent merge:

uiowa06 - 256361700
uiowa07 - 256361704

First part of uiowa06:
```
2024-10-08T16:38:27Z Deploying main-build to uiowa06stg
2024-10-08T16:41:28Z main-build deployed to uiowa06stg
2024-10-08T16:43:04Z Starting hook: post-code-update
2024-10-08T16:43:04Z Executing: "/var/www/html/hooks/test/post-code-update/post-code-update.sh" "uiowa06" "test" main-build main-build uiowa06@svn-45062.prod.hosting.acquia.com:uiowa06.git git < /dev/null (as uiowa06.test)
2024-10-08T16:43:04Z site="$1"
2024-10-08T16:43:04Z target_env="$2"
2024-10-08T16:43:04Z source_branch="$3"
2024-10-08T16:43:04Z deployed_tag="$4"
2024-10-08T16:43:04Z repo_url="$5"
2024-10-08T16:43:04Z repo_type="$6"
2024-10-08T16:43:04Z # Prep for BLT commands.
2024-10-08T16:43:04Z repo_root="/var/www/html/$site.$target_env"
2024-10-08T16:43:04Z export PATH=$repo_root/vendor/bin:$PATH
2024-10-08T16:43:04Z cd $repo_root
2024-10-08T16:43:04Z blt artifact:ac-hooks:post-code-update $site $target_env $source_branch $deployed_tag $repo_url $repo_type --environment=$target_env --no-interaction -D drush.ansi=false
2024-10-08T16:43:04Z Running updates for environment: test
2024-10-08T16:43:04Z > artifact:update:drupal:all-sites
2024-10-08T16:43:04Z Running multisite updates in parallel.
2024-10-08T16:43:04Z  [Exec] Running parallel -j 3 blt uiowa:site:update ::: 'medicine.uiowa.edu' 'acb.medicine.uiowa.edu' 'anesthesia.medicine.uiowa.edu' 'appil.medicine.uiowa.edu' 'biochemstores.medicine.uiowa.edu' 'bioethics.medicine.uiowa.edu' 'biomed.medicine.uiowa.edu' 'bmb.medicine.uiowa.edu' 'cardiovascular.medicine.uiowa.edu' 'cbio.medicine.uiowa.edu' 'cme.medicine.uiowa.edu' 'community.medicine.uiowa.edu' 'cpss.uihealthcare.org' 'craniofacial.medicine.uiowa.edu' 'crystallography.medicine.uiowa.edu' 'ct-surgery.medicine.uiowa.edu' 'demo.medicine.uiowa.edu' 'dermatology.medicine.uiowa.edu' 'diabetes.medicine.uiowa.edu' 'emergencymedicine.medicine.uiowa.edu' 'eye.medicine.uiowa.edu' 'facultyaffairs.medicine.uiowa.edu' 'familymedicine.medicine.uiowa.edu' 'flowcytometry.medicine.uiowa.edu' 'genetherapy.medicine.uiowa.edu' 'geneticscluster.medicine.uiowa.edu' 'genomeediting.medicine.uiowa.edu' 'humangenetics.medicine.uiowa.edu' 'icts.uiowa.edu' 'immunologycenter.medicine.uiowa.edu' 'inbc.medicine.uiowa.edu' 'inflammation.medicine.uiowa.edu' 'internalmedicine.medicine.uiowa.edu' 'iowalionseyebank.org' 'iowaneuroscience.medicine.uiowa.edu' 'it.medicine.uiowa.edu' 'kidsight.medicine.uiowa.edu' 'levittcenter.medicine.uiowa.edu' 'md.medicine.uiowa.edu' 'microbiology.medicine.uiowa.edu' 'mls.medicine.uiowa.edu' 'mmed.medicine.uiowa.edu' 'mri.medicine.uiowa.edu' 'mstp.medicine.uiowa.edu' 'ncbc.medicine.uiowa.edu' 'neurology.medicine.uiowa.edu' 'neuroscience-pharmacology.medicine.uiowa.edu' 'neurosurgery.medicine.uiowa.edu' 'nfrc.medicine.uiowa.edu' 'nmr.medicine.uiowa.edu' 'obesity.medicine.uiowa.edu' 'obgyn.medicine.uiowa.edu' 'ocrme.medicine.uiowa.edu' 'ogps.medicine.uiowa.edu' 'orthopedics.medicine.uiowa.edu' 'oscep.medicine.uiowa.edu' 'otolaryngology.medicine.uiowa.edu' 'pa.medicine.uiowa.edu' 'pain.medicine.uiowa.edu' 'pathology.medicine.uiowa.edu' 'pbi.medicine.uiowa.edu' 'pediatrics.medicine.uiowa.edu' 'perfusion.medicine.uiowa.edu' 'physiology.medicine.uiowa.edu' 'ponseti.medicine.uiowa.edu' 'proteomics.medicine.uiowa.edu' 'pstp.medicine.uiowa.edu' 'psychiatry.medicine.uiowa.edu' 'pt.medicine.uiowa.edu' 'radiationoncology.medicine.uiowa.edu' 'radiology.medicine.uiowa.edu' 'radsci.medicine.uiowa.edu' 'research.medicine.uiowa.edu' 'respiratorycareseminar.uihealthcare.org' 'saic.medicine.uiowa.edu' 'sercc.medicine.uiowa.edu' 'shpep.medicine.uiowa.edu' 'surgery.medicine.uiowa.edu' 'uidl.medicine.uiowa.edu' 'urology.medicine.uiowa.edu' 'vaccine.medicine.uiowa.edu' 'vectorcore.medicine.uiowa.edu' 'wellstone.medicine.uiowa.edu' 2>&1 | tee -a /shared/logs/parallel_deploy_log_uiowa06_20241008_164304.log
```

All of of uiowa07:
```
2024-10-08T16:38:29Z Deploying main-build to uiowa07stage
2024-10-08T16:40:32Z main-build deployed to uiowa07stage
```

**Comment by e-marie-w** [comment]
Awesome!! I noticed that uiowa07 only has that one commit while apps < uiowa07 have two commits, is that related/relevant?

**Comment by joewhitsitt** [comment]
Good eye. Yeah, you could probably include that info in the ticket

**Comment by e-marie-w** [comment]
Ticket submitted!

**Comment by joewhitsitt** [comment]
Looks like we can duplicate the test directory as `stage` and things will just work. 🤷 

hooks/test copy to hooks/stage
